src/transformers/generation_tf_utils.py

```diff
@@ -1761,11 +1492,6 @@ def _generate(
                 input_ids, return_dict_in_generate, model_kwargs
             )
 
-        # TODO(Patrick) - ugly `past`/`encoder_output` hack here which requires a bigger
-        # refactor of all generation models in TF. `past` should be
-        # optional everywhere and not be set equal to encoder_outputs
-        model_kwargs["past"] = model_kwargs.get("encoder_outputs")[:1] if self.config.is_encoder_decoder else None
```

Comment:
This was moved a few lines below, as it needed to be called after `_expand_inputs_for_generation()` in the sample generate mode